### PR TITLE
Moving QML rendering off the main thread

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: be5ef2bedda3ef9c438520dd41bdd889917b302c available at: https://gist.github.com/c98f3239634ed7c3f55a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6613-win32.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6613.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 08f68d695c6029c6f3fe619d801155a7711a1225 available at: https://gist.github.com/c98f3239634ed7c3f55a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6613-win32.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6613.dmg

File: libraries/gl/src/gl/GLEscrow.h [hunk 1]
```diff
@@ -154,35 +178,55 @@ class GLEscrow {
...
+                    _trash.push_front(_submits.front());
+                    _submits.pop_front();
+                    trashed = true;
+                    ++wastedWork;
+                }
+            });
+        }
+            
+        trashed = true;
+        while (trashed) {
```
Comment (ZappoMan) [inline libraries/gl/src/gl/GLEscrow.h]: sounds like tuesday night

File: libraries/gl/src/gl/OffscreenQmlSurface.cpp [hunk 2]
```diff
@@ -241,50 +226,44 @@ class OffscreenQmlRenderer : public OffscreenGLCanvas {
...
-            // FIXME The web browsers seem to be leaving GL in an error state.
-            // Need a debug context with sync logging to figure out why.
-            // for now just clear the errors
-            glGetError();
+            PROFILE_RANGE("qml_render")
+            TexturePtr texture = _textures.getNextTexture();
+            _fbo->Bind(Framebuffer::Target::Draw);
+            _fbo->AttachTexture(Framebuffer::Target::Draw, FramebufferAttachment::Color, *texture, 0);
+            _fbo->Complete(Framebuffer::Target::Draw);
+            //Context::Clear().ColorBuffer();
```
Comment (ZappoMan) [inline libraries/gl/src/gl/OffscreenQmlSurface.cpp]: dead?

**Comment by hifi-gustavo** [comment]
Build console output for commit: c6306c4eb3f829d00b6cc472ac8de5974c57a806 available at: https://gist.github.com/c98f3239634ed7c3f55a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6613-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6613.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae1345bb13275e4e7cb2f0bd84ff08c09b0c5696 available at: https://gist.github.com/c98f3239634ed7c3f55a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6613-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6613.dmg

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae1345bb13275e4e7cb2f0bd84ff08c09b0c5696 available at: https://gist.github.com/c98f3239634ed7c3f55a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6613-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6613.dmg

**Pull request merged**